### ISSUE TITLE
False positive: unused_declaration flagging projectedValue of property wrapper

### ISSUE THREAD

**Issue by rebello95** [realm/SwiftLint#3237]
### Describe the bug

Swiftlint is identifying the `projectedValue` of a property wrapper as unused.

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.39.2
* Installation method used (Homebrew, CocoaPods, building from source, etc)? Homebrew

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? No
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 11.3.1 Build version 11C505
* Do you have a sample that shows the issue? 

```swift
@propertyWrapper
final class LockedWhenRead<Value: Equatable> {
    /// Allows access outside of private scope.
    var projectedValue: LockedWhenRead<Value> { return self }

    var wrappedValue: Value {
        get { self.getAndLockValue() }
        set { self.storeAndSet(newValue) }
    }

    private func getAndLockValue() -> Value {
        ...
    }

    private func storeAndSet(_ newValue: Value) {
        ...
    }
}
```

**Comment by keith** [comment]
Still relevant

**Comment by iressler** [comment]
Bumping this because I just ran into this too.
